cajita/src/Cajita_FastFourierTransform.hpp

```diff
@@ -407,6 +420,40 @@ struct HeffteScalingTraits<FFTScaleSymmetric>
 {
     static const auto scaling_type = heffte::scale::symmetric;
 };
+
+#ifdef KOKKOS_ENABLE_SYCL
+// Overload for SYCL.
+template <class HeffteBackendType>
+auto createHeffte(
+    HeffteBackendType, heffte::box3d<> inbox, heffte::box3d<> outbox,
+    MPI_Comm comm, heffte::plan_options params,
+    typename std::enable_if<
+        std::is_same<HeffteBackendType, heffte::backend::onemkl>::value,
+        int>::type* = 0 )
+{
+    // Set FFT options from given parameters
+    // heFFTe correctly handles 2D or 3D FFTs within "fft3d"
+    // FIXME: get SYCL queue from Kokkos.
+    sycl::queue q;
```

Comment:
This is illegal!

The queue has local scope, but heFFTe needs to keep a reference to the queue for all operations. This is needed for synchronization and management of devices on platforms with multiple GPUs. The correct course of action is to use the Kokkos global sycl queue.

Reply:
Thanks! Updated locally - @masterleinad am I correct that we need Kokkos >=3.7 in order to get the SYCL queue?

Reply:
> Thanks! Updated locally - @masterleinad am I correct that we need Kokkos >=3.7 in order to get the SYCL queue?

Yes, for
```C++
sycl::queue& q = space.sycl_queue();
```
but you could do
```C++
 Kokkos::Experimental::Impl::SYCLInternal& instance =
        *space.impl_internal_space_instance();
    sycl::queue& q = *instance.m_queue;
```
(which is of course discouraged) even before.